### DISCUSSION TITLE
Missing `Checks` permission in personal access token

### DISCUSSION THREAD

**Discussion by kemchenj** [community/community#129512] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

There is `Checks` repository permission that listed in 《[Permissions required for fine-grained personal access tokens](https://docs.github.com/en/rest/authentication/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28#repository-permissions-for-checks)》

<img width="631" alt="Screenshot 2024-06-23 at 22 26 26" src="https://github.com/community/community/assets/11192474/c2eacbdb-db25-4c41-ae60-d7f0a520c470">

But it is missing in the [settings](https://github.com/settings/tokens?type=beta)：

<img width="444" alt="Screenshot 2024-06-23 at 22 27 36" src="https://github.com/community/community/assets/11192474/11ef71e2-5137-4314-8b39-45b718f1e9a8">

I wonder if this is a bug or it is intended?

**Comment by froblesmartin** [comment]
I have just faced this as well. Curious that, as you mention, it is documented in the fine grained page, but not available. Checking in a GitHub App, it does show it:

![image](https://github.com/user-attachments/assets/615874af-c1bd-4d90-9c5f-2ca8c2d2386c)

**Reply by richardsimko** [reply]
@kemchenj or @froblesmartin Did you figure out some workaround? Seems to still be an issue.

**Reply by PhilipAtCisco** [reply]
++ I'm also running into this myself this morning trying to setup some CI tests that create PRs to test a reuseable workflow.

PRs created by bots don't run checks (https://github.com/aws/karpenter-provider-aws/issues/4525) so a fine-grained PAT that is repo scoped I believe is the workaround.  I think I could swap back to the regular GITHUB_TOKEN for the validation of the checks, but this missing feature would certainly simplify things!

**Reply by richardsimko** [reply]
Yeah we're in a similar situation but With Renovate. Had to go back to classic tokens.

**Reply by FWBlueWizard** [reply]
Also now running into this as well. Started to lose my mind how hard it was to find this related issue. Glad it it's here 
++++

**Comment by iufor** [comment]
I have same issue facing

**Comment by andrewlarioza** [comment]
We stumble to this issue as well and good to have the resolution soon.
`$ curl -L   -H "Accept: application/vnd.github+json"   -H "Authorization: Bearer github_pat_XXXXX"   -H "X-GitHub-Api-Version: 2022-11-28"   https://api.github.com/repos/azoiral/devops-test-repo/commits/DEVOPS-3-gatekeeper-v1-3/check-suites
{
  "message": "Resource not accessible by personal access token",
  "documentation_url": "https://docs.github.com/rest/checks/suites#list-check-suites-for-a-git-reference",
  "status": "403"
}`

**Comment by andipexton-insight** [comment]
To spend hours building a process based on documented functionality that doesn't exist is more than frustrating, it would be nice if someone from @GitHub could give a response? is the error in the documentation or the omission of the feature?

**Reply by FWBlueWizard** [reply]
Great question. Same problem.

**Comment by omarabid** [comment]
So anyone from Github care to comment or look at this. I spent a whole morning trying to figure out why my "fine-tuned" tokens do not work and it turns out that there was no "Checks" fine tuning.

**Comment by frankalbenesius** [comment, marked as answer]
Just in case anyone else finds this thread, here was a response I got from emailing support:

>I apologize for any inconvenience here, but at the moment, it isn't possible to assign Checks permissions to a Fine-grained PAT —only GitHub Apps can access this API. We had that functionality initially, but due to some edge cases we disabled it.
>
> Our engineering team have an issue tracking this and we've added your interest to the issue. In future, we may re-introduce the ability to configure Fine-grained PATs with permissions to run Checks API. When we do publish it, we'll make sure to document it in the [GitHub Changelog](https://github.blog/changelog/) or the [GitHub Blog](https://github.blog/).
>
> As an alternative I would suggest to make use of the[ GitHub App](https://docs.github.com/en/enterprise-cloud@latest/apps/creating-github-apps/about-creating-github-apps) to query the required API endpoints. If you're using Actions as your CI, you can [authenticate Actions as a GitHub App](https://docs.github.com/en/enterprise-cloud@latest/apps/creating-github-apps/authenticating-with-a-github-app/making-authenticated-api-requests-with-a-github-app-in-a-github-actions-workflow).


Summary: Use a GitHub App instead of a PAT.

**Reply by Lewenhaupt** [reply]
Classic tokens also work!

**Comment by voodoodror** [comment]
Following check-runs classic token modification deprecation https://github.blog/changelog/2025-02-12-notice-of-upcoming-deprecations-and-breaking-changes-for-github-actions/#changes-to-check-run-status-modification
> Check run status and conclusions can only be updated internally by GitHub Actions.

The documentation says that [fine-grained tokens are supported](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#update-a-check-run). Thankfully, I found this issue, and it saved me time from debugging it. I wonder why they wrote it in the docs in the first place if they know it's not supported.

**Reply by aredridel** [reply]
I have a strong suspicion that Copilot wrote the docs and nobody checked. That AI mandate bears this fruit often.

**Comment by NedWilbur** [comment]
It seems the official API docs are ahead of what [is actually supported today](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens-limitations)... 

API Docs - https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#get-a-check-run
<img width="613" height="644" alt="image" src="https://github.com/user-attachments/assets/66f3a952-4551-42a9-bfac-63917c61490e" />

<img width="748" height="473" alt="image" src="https://github.com/user-attachments/assets/f13fadba-14f3-45b7-953b-a5d1b6eb32cc" />

**Comment by phenggeler** [comment]
It sounds like Github is planning to [deprecate classic tokens](https://github.blog/security/supply-chain-security/our-plan-for-a-more-secure-npm-supply-chain/). The community will need Checks to be supported for fine-grained access tokens before that happens. Would love to see progress on this by Github

**Reply by valentjn** [reply]
The linked info is about npm classic tokens. Those have indeed been deprecated and in the meantime [permanently revoked on December 9, 2025](https://github.blog/changelog/2025-12-09-npm-classic-tokens-revoked-session-based-auth-and-cli-token-management-now-available/#npm-classic-tokens-permanently-revoked).

GitHub classic PATs are a completely different thing. While [GitHub recommends using fine-grained PATs over classic PATs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#types-of-personal-access-tokens), classic PATs are currently not deprecated since GitHub has a [list of limitations of fine-grained PATs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens-limitations) they are working on. I think GitHub won't deprecate classic PATs until most of that list has been resolved.

**Comment by schpet** [comment]
i'm interested in using fine grained tokens to give agents isolated access to repos. but with this limitation they cannot see if the CI checks pass on github actions.

**Comment by karszawa** [comment]
We are also affected by this limitation. We use fine-grained PATs to give [Claude Code](https://docs.anthropic.com/en/docs/claude-code) (Anthropic's AI coding agent) access to our private organization repositories.

Claude Code's web-based environment ([Claude Code on claude.ai](https://docs.anthropic.com/en/docs/claude-code/claude-code-on-claude-ai)) relies on fine-grained PATs for GitHub authentication. In this environment, `gh pr view` always fails with the `statusCheckRollup` error because there is no way to grant the Checks permission to the token.

As AI-powered coding agents like Claude Code become more widely adopted, the demand for fine-grained PATs with proper Checks read access is growing rapidly. These agents need to read PR status checks to understand CI results, review PRs effectively, and assist developers in their workflows. The `--json` workaround exists, but it is not ideal — the default `gh pr view` experience should just work.

With the [planned deprecation of classic PATs](https://github.blog/security/supply-chain-security/our-plan-for-a-more-secure-npm-supply-chain/), this gap will become a blocker for many teams using AI coding tools in their development workflow.

**Environment:** `gh` v2.86.0 on Claude Code Web, fine-grained PAT scoped to an organization's private repository.

**Reply by smartcontracts** [reply]
Yeah this is a massive blocker for secure Claude Code usage.

**Reply by azu** [reply]
It seems the Codex App also depends on the Check API via the `gh` command in the same way, so it appears it can't be used with a Fine-grained PAT.

**Reply by EugenMayer** [reply]
The above sums it up fairly well. 2 years later  the corner cases should have been fixed, shouldnt those?

**Comment by davidawad** [comment]
I had no clue this was an issue and wasted a good chunk of time and money on it as I thought I could automatically get pull request status via the api. Is there a reason for the limitation? 

thanks

**Comment by elvisun** [comment]
I ended up working around it by hacking my own `gh pr checks` impl

```
 # Override gh to fix fine-grained PAT limitation with Checks API
 gh() {
   # Intercept: gh pr checks [PR_NUM] [flags]
   if [[ "$1" == "pr" && "$2" == "checks" ]]; then
     shift 2  # remove "pr checks"
     _gh_pr_checks "$@"
     return $?
   fi
   # Pass through all other gh commands
   command gh "$@"
 }

 _gh_pr_checks() {
   local pr_num="$1"
   local repo
   repo=$(command gh repo view --json nameWithOwner -q '.nameWithOwner' 2>/dev/null)

   # If no PR number, get from current branch
   if [[ -z "$pr_num" || "$pr_num" == -* ]]; then
     pr_num=$(command gh pr view --json number -q '.number' 2>/dev/null)
     if [[ -z "$pr_num" ]]; then
       echo "No PR found for current branch" >&2
       return 1
     fi
   fi

   # Get branch from PR
   local branch
   branch=$(command gh api "repos/$repo/pulls/$pr_num" --jq '.head.ref' 2>/dev/null)

   # Get latest workflow run
   local run_data
   run_data=$(command gh api "repos/$repo/actions/runs?branch=$branch&per_page=5" \
     --jq '.workflow_runs | map(select(.event == "pull_request")) | .[0] | {id, name, status, conclusion, html_url}' 2>/dev/null)

   local run_id=$(echo "$run_data" | jq -r '.id')

   if [[ -z "$run_id" || "$run_id" == "null" ]]; then
     echo "No workflow runs found for PR #$pr_num" >&2
     return 1
   fi

   echo "PR #$pr_num ($branch)"
   echo ""

   # Get jobs and format like gh pr checks
   command gh api "repos/$repo/actions/runs/$run_id/jobs" --jq '
     .jobs[] |
     (if .conclusion == "success" then "✓ pass"
      elif .conclusion == "failure" then "X fail"
      elif .conclusion == "skipped" then "- skip"
      elif .status == "in_progress" then "* runs"
      elif .status == "queued" then "- pend"
      else "- " + .status end) + "\t" + .name + "\t" + .html_url
   ' 2>/dev/null | column -t -s $'\t'

   # Return appropriate exit code
   local has_failure=$(command gh api "repos/$repo/actions/runs/$run_id/jobs" --jq '[.jobs[].conclusion] | any(. == "failure")' 2>/dev/null)
   local has_pending=$(command gh api "repos/$repo/actions/runs/$run_id/jobs" --jq '[.jobs[].status] | any(. == "in_progress" or . == "queued")' 2>/dev/null)

   if [[ "$has_failure" == "true" ]]; then
     return 1
   elif [[ "$has_pending" == "true" ]]; then
     return 2
   fi
   return 0
 }
```

**Comment by fipster-dev** [comment]
2 years after first report and this is still an issue... how embarrassing!

**Comment by dfinlay** [comment]
Still an issue, it would be really quite nice if this got resolved. The Checks API already has support for application permissions, so it hopefully shouldn't be too hard to add it to the Fine Grained PAT permission builder. (Assuming the documentation is correct and the API endpoints already support fine grained PAT and just the permission builder is missing the functionality). Technically the Checks API docs may be correct, but then you've locked the Checks API in a manner that is impossible to make a functional key for.

### Mismatched API Docs

The Checks API Documentation for [Create a Check Run](https://docs.github.com/en/rest/checks/runs?apiVersion=2026-03-10#get-a-check-run--parameters) 

<img width="465" height="222" alt="ScreenFloat Shot of Google Chrome Beta on 2026-08-05 at 15-06-29" src="https://github.com/user-attachments/assets/9ee18ba3-190e-4929-9d4c-1cbd27e992aa" />

and [Get a Check Run ](https://docs.github.com/en/rest/checks/runs?apiVersion=2026-03-10#get-a-check-run--parameters) state that fine grained access PATs work with the Check API.

<img width="455" height="320" alt="ScreenFloat Shot of Google Chrome Beta on 2026-08-05 at 15-07-15" src="https://github.com/user-attachments/assets/98b456e7-7d95-4370-8235-874a37f30e82" />


_However_ the [PAT limitation documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens-limitations) says that calling Check API is a known gap. So there is a contradiction here, either the limitation is in the Checks API[^1] OR it's simply in the fine grained permission builder. 

<img width="767" height="455" alt="ScreenFloat Shot of Google Chrome Beta on 2026-08-05 at 15-20-23" src="https://github.com/user-attachments/assets/d88d67d1-2d31-45a5-9f09-e86fd19de2da" />

[^1]: If the Checks API doesn't support fine grained PATs, it would make sense that the permission builder doesn't include it. But it claims to.

**Comment by tzarebczan** [comment]
Ran into this also..come on github, you can do better!

**Comment by larrasket** [comment]
it's terrible this is not fixed yet

**Comment by bart-g** [comment]
I am also blocked by this 👎
